### ISSUE TITLE
Syntax Highlighting Error

### ISSUE THREAD

**Issue by JonMcPhalen** [parallaxinc/Propeller-Tool#152]
This has been an ongoing issue: on occasion the syntax highlighting feature fails and causes a large chunk of the editor display to appear to be selected, except that the selection color is white. Switching back and forth between editor display modes will usually fix the screen.

![image](https://user-images.githubusercontent.com/4042568/128751959-8d2c93ab-67d9-4597-bca8-18115d39e87e.png)

**Comment by PropGit** [comment]
Does this only happen to you on the Summary view, or other views as well?

**Comment by JonMcPhalen** [comment]
I think it's all views (I only uses Full Source and Summary), but I cannot say with certainly. I will pay attention to conditions each time this happens in the future.

**Comment by JonMcPhalen** [comment]
Still happens in 2.6.0 -- but only the first time Summary mode is selected after opening the file.

**Comment by PropGit** [comment]
Yes, I still see this once in a while also.  Seems to happen consistently with specific files and not with others.

For me, 

- Open file, toggle back and forth between Summary and Full Source = continues to exhibit the problem in Summary view
- Open file, switch to Summary, click to set the cursor anywhere else within the source code, toggle to Full Source then back to Summary = problem goes away
- Open file, switch to Summary, press down arrow key to move the cursor one line down within the source code, toggle to Full Source then back to Summary = problem goes away
- Open file, press down arrow key to move the cursor one line down within the source code, toggle to Full Source, Summary, Full Source again = problem goes away

Because the problem isn't static (goes away) without changing the source code itself, I don't think it's a syntax rule issue solely but more likely an initialization problem related to syntax rules and that specific view mode.

I also just noticed that if you scroll down (in Full Source or Condensed) view, as long as you've scrolled down enough (not necessarily to the method where the problem occurs) then stitch to Summary view, the problem will still be there but will start on a later line (i.e. it happens on fewer lines of the Summary display).  This last symptom really does sound like something in the syntax highlight buffer is not being initialized properly upon file load and it being refreshed (up to a point) by the scroll action or cursor movement.  That sounds like a good clue for me to follow in the code.
